books.c

```diff
@@ -146,7 +180,84 @@ Book* filterBooks(Book* parameters, Book* head){
     if (parameters->ISBN) filtered_books = filterByISBN(parameters->ISBN, filtered_books);
     if (parameters->title) filtered_books = filterByTitle(parameters->title, filtered_books);
     if (parameters->total_count > 0) filtered_books = filterByTotalCount(parameters->total_count, filtered_books);
+    if (parameters->tag_count > 0)   filtered_books = filterByTags(parameters->tags, parameters->tag_count, filtered_books);
     return filtered_books;
+}
+
+BookCopy* createBookCopy(int book_uid, char* ISBN, enum book_status status)
+{
+    BookCopy* book_copy = malloc(sizeof(BookCopy));
+    if (!book_copy)
+    {
+        printf("Fail to allocate memory for book.\n");
+        return NULL;
+    }
+    book_copy->book_uid = book_uid;
+    book_copy->ISBN = ISBN;
+    book_copy->status = status;
+    book_copy->next = NULL;
+    return book_copy;
+}
+
+void insertBookCopy(BookCopy* book_copy, BookCopy** head)
+{
+    if (!*head)
+    {
+        *head = book_copy;
+        return;
+    }
+    BookCopy* current = *head;
+    while (current->next)
+        current = current->next;
+    current->next = book_copy;
+}
+
+void printBookCopies(BookCopy* head)
```

Comment:
I think we have to create similar functions for different kind of structs since we can't use a prototype function for different structs in C. We discussed about this when we tried to use default parameters.